packages/share/ios/Classes/SharePlugin.m

```diff
@@ -16,8 +16,9 @@ + (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar> *)registrar {
   [shareChannel setMethodCallHandler:^(FlutterMethodCall *call, FlutterResult result) {
     if ([@"share" isEqualToString:call.method]) {
       NSDictionary *arguments = [call arguments];
+      NSString *shareText = arguments[@"text"];
 
-      if ([arguments[@"text"] length] == 0) {
+      if (shareText.length == 0) {
```

Comment:
You might consider removing this check too. It is already checked on the Dart side. Or we can leave that for another PR. You call.

Reply:
@mravn-google i deliberately left that logic as is - just replaced `arguments[@"text"]` with the unpacked `shareText` variable. That line is the main reason I opted for the `if ([arguments[@"text"] isKindOfClass:[NSString class]] == YES) {` type check, because it could throw an exception if `shareText` is not `nil` or an `NSString` (for example: `[NSNull null]`).